src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const initialStateClone = { ...state };
+
+  const clonedStates = [];
+
+  for (const action of actions) {
+    const currentStateClone = clonedStates.length > 0
+      ? { ...clonedStates[clonedStates.length - 1] }
+      : { ...initialStateClone };
+
+    switch (action.type) {
+      case 'addProperties':
```

Comment:
Describe all variants of cases in constant.
const ACTION_CASES = {
add: 'addProperties',
}